docker-compose.yml

```diff
@@ -1,63 +1,91 @@
-# This file is used for pulling Covalent images from ECR using docker-compose
-# Example usage: Run "docker-compose --env-file .env.compose up" from the root of the project
-# to pull the images from ECR and start the containers.
+# This file is used for local development with docker-compose and for building microservice images in the CD pipeline.
+# Example Usage: Run "docker-compose -f docker-compose.yml up --build" from the root of the project
+# to build the images using this file and start the containers.
 
 version: "3.9"
 
 services:
   covalent-data:
-    image: "public.ecr.aws/covalent/covalent_data"
+    build:
+      context: .
+      dockerfile: refactor/data/Dockerfile
+    image: covalent-data
+    command: sh -c "python main.py"
     ports:
-      - "80:8004"
+      - "8004:8004"
     working_dir: /data
     networks:
       covalent:
 
   covalent-dispatcher:
-    image: "public.ecr.aws/covalent/covalent_dispatcher"
+    build:
+      context: .
+      dockerfile: refactor/dispatcher/Dockerfile
+    image: covalent-dispatcher
+    command: sh -c "python main.py"
     ports:
-      - "80:8002"
+      - "8002:8002"
     working_dir: /dispatcher
     networks:
       covalent:
 
   covalent-queuer:
-    image: "public.ecr.aws/covalent/covalent_queuer"
+    build:
+      context: .
+      dockerfile: refactor/queuer/Dockerfile
+    image: covalent-queuer
+    command: sh -c "python main.py"
     ports:
-      - "80:8001"
+      - "8001:8001"
     working_dir: /queuer
     networks:
       covalent:
 
   # Queue-consumer uses the same Dockerfile configurations as the Queuer
   covalent-queue-consumer:
-    image: "public.ecr.aws/covalent/covalent_queuer"
+    build:
+      context: .
+      dockerfile: refactor/queuer/Dockerfile
+    image: covalent-queue-consumer
+    command: sh -c "python main.py"
     ports:
-      - "80:8001"
+      - "3000:8001"
```

Comment:
does this service need to expose a port?

Reply:
I'm not sure... @AlejandroEsquivel @Emmanuel289 does this look right for the consumer?

Reply:
@scottwn I exposed port 8001, similar to the queuer. Maybe @AlejandroEsquivel can confirm if this is correct for the service.

Reply:
@scottwn @Emmanuel289 the queue consumer does not expose an API, it just connects to NATS so we do not require port mappings here